core/src/test/java/nl/nn/adapterframework/util/CompactSaxHandlerTest.java

```diff
@@ -106,75 +110,112 @@ void testElementToMoveFeatureSessionKey() throws IOException, SAXException {
-		handler = new CompactSaxHandler();
+		handler = new CompactSaxHandler(new XmlWriter());
 		handler.setChompLength(80);
 		handler.setRemoveCompactMsgNamespaces(true);
 		handler.setContext(session);
 		handler.setElementToMoveChain("Envelope;Body;edcLk01;object;identificatie");
 		XmlUtils.parseXml(testOutputFile, handler);
 
 		// Assert 2: everything should be still the same
-		assertEquals(testOutputFile, handler.getXmlString());
-		assertEquals("DC2023-00020", session.get("ref_identificatie"));
-		assertEquals("DC2022-012345", session.get("ref_identificatie2"));
+		assertEquals(testOutputFile, handler.toString());
+		assertEquals("DC2023-00020", session.getString("ref_identificatie"));
+		assertEquals("DC2022-012345", session.getString("ref_identificatie2"));
+	}
+
+	@Test
+	void testElementToMoveBigData() throws IOException, SAXException {
+		// Arrange
+		handler.setChompLength(200_000);
+		handler.setRemoveCompactMsgNamespaces(true);
+		handler.setElementToMove("message");
+		String expectedOutput = "<event timestamp=\"0\" level=\"DEBUG\">\n" +
+				"  <message>{sessionKey:ref_message}</message>\n" +
+				"</event>";
+
+		// Act
+		Message bigInputMessage = MessageTestUtils.getMessage("/Logging/xml-of-pdf-file.log");
+		XmlUtils.parseXml(bigInputMessage.asInputSource(), handler);
+
+		// Assert
+		assertEquals(expectedOutput, handler.toString());
+		assertEquals(101_541, session.getString("ref_message").length());
 	}
 
 	@Test
 	void testElementToMoveChainOnlyRightLocation() throws IOException, SAXException {
 		// Arrange
 		handler.setChompLength(80);
 		handler.setRemoveCompactMsgNamespaces(true);
-		PipeLineSession session = new PipeLineSession();
-		handler.setContext(session);
 		handler.setElementToMoveChain("Envelope;Body;edcLk01;object;identificatie");
 
 		// Act
 		XmlUtils.parseXml(MessageTestUtils.getMessage("/Util/CompactSaxHandler/input-chaintest.xml").asInputSource(), handler);
 
 		// Assert
 		String testOutputFile = TestFileUtils.getTestFile("/Util/CompactSaxHandler/output-chaintest2.xml");
-		assertEquals(testOutputFile, handler.getXmlString());
-		assertEquals("DC2023-00020", session.get("ref_identificatie"));
-		assertEquals("DC2022-012345", session.get("ref_identificatie2"));
+		assertEquals(testOutputFile, handler.toString());
+		assertEquals("DC2023-00020", session.getString("ref_identificatie"));
+		assertEquals("DC2022-012345", session.getString("ref_identificatie2"));
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings = {
+			"/Util/CompactSaxHandler/input-invalid-moved-ref-1.xml",
+			"/Util/CompactSaxHandler/input-invalid-moved-ref-2.xml",
+			"/Util/CompactSaxHandler/input-invalid-moved-ref-3.xml",
+			"/Util/CompactSaxHandler/input-invalid-moved-ref-4.xml",
+			"/Util/CompactSaxHandler/input-invalid-moved-ref-5.xml",
+			"/Util/CompactSaxHandler/input-invalid-moved-ref-6.xml"
+	})
+	void testElementToMoveEdgeCases(String inputUri) throws IOException, SAXException {
```

Comment:
Ik snap nog niet helemaal wat je hier tests? De input is overal verkeerd? Test je of hij de inhoud van het `identificatie` element zet (al staan er rare characters in?) ?

Reply:
Tests die ik had toegevoegd voor de refactoring, om er zeker van te zijn dat "verkeerde" input niet leid tot nog verkeerdere output.

Reply:
Dan is het misschien handig een check te doen op de session, dat daar niets bij komt bijvoorbeeld.

Reply:
Prima hoor. Toegevoegd.